python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -173,6 +174,26 @@ private void addSymbolsToTree(FileInputImpl fileInput) {
     }
   }
 
+  private static void handleClassMemberAccess(FileInput fileInput) {
+    BaseTreeVisitor classMemberAccess = new BaseTreeVisitor() {
+      @Override
+      public void visitQualifiedExpression(QualifiedExpression qualifiedExpression) {
+        super.visitQualifiedExpression(qualifiedExpression);
+        if (qualifiedExpression.qualifier() instanceof HasSymbol) {
+          Symbol symbol = ((HasSymbol) qualifiedExpression.qualifier()).symbol();
+          if (symbol != null && symbol.kind() == Symbol.Kind.CLASS) {
+            Symbol member = ((ClassSymbolImpl) symbol).membersByName().get(qualifiedExpression.name().name());
+            if (member != null) {
+              Optional.ofNullable(qualifiedExpression.name().usage())
+                .ifPresent(usage -> ((SymbolImpl) member).addUsage(qualifiedExpression.name(), usage.kind()));
```

Comment:
There's something I don't get. Are we relying on the fact that `qualifiedExpression.name()` was associated to a wrong symbol to reassign it to the correct one?

Reply:
I agree that's not ideal: I'm relying on it to get the correct usage kind. I will change this